src/constant.h

```diff
@@ -7,594 +7,598 @@
 // Global constants used by Moria
 // clang-format off
 
-#define MAX_UCHAR         255
-#define MAX_SHORT       32767   // maximum short/long signed ints
-#define MAX_LONG   0x7FFFFFFFL
+#pragma once
+
+#include "headers.h"
```

Comment:
The problem is, that std::numeric_limits is contained in <limits>. So its necessary that <limits> is included in this header. All other includes from standard library et cetera are in "headers.h", so I included it here. We should find a better way.